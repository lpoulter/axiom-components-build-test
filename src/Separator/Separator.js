import PropTypes from 'prop-types';
import classnames from 'classnames';
import React, { Component } from 'react';
import Base from '../Base/Base';
import './Separator.css';

export default class Separator extends Component {
  static propTypes = {
    borderStyle: PropTypes.oneOf(['solid', 'dotted']),
    className: PropTypes.string,
    direction: PropTypes.oneOf(['horizontal', 'vertical']),
  };

  static defaultProps = {
    borderStyle: 'solid',
    direction: 'horizontal',
  };

  render() {
    const { borderStyle, direction, className, ...rest } = this.props;
    const classes = classnames('ax-separator',
      `ax-separator--${borderStyle}`,
      `ax-separator--${direction}`,
      className
    );

    return (
      <Base
          space="x0"
          { ...rest }
          Component="hr"
          className={ classes } />
    );
  }
}

