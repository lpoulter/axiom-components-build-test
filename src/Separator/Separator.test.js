import React from 'react';
import renderer from 'react-test-renderer';
import Separator from './Separator';

const getComponent = (props = {}) =>
  renderer.create(
    <Separator { ...props } />
  );

describe('Separator', () => {
  it('renders with defaultProps', () => {
    const component = getComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('accepts className prop', () => {
    const component = getComponent({ className: 'test' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('renders with style', () => {
    ['solid', 'dotted'].forEach((borderStyle) => {
      it(borderStyle, () => {
        const component = getComponent({ borderStyle });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });

  describe('renders with direction', () => {
    ['horizontal', 'vertical'].forEach((direction) => {
      it(direction, () => {
        const component = getComponent({ direction });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
