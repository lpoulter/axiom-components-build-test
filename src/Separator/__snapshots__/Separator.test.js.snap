// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Separator renders with defaultProps 1`] = `
<hr
  className="ax-separator ax-separator--solid ax-separator--horizontal ax-space--x0"
/>
`;

exports[`Separator accepts className prop 1`] = `
<hr
  className="ax-separator ax-separator--solid ax-separator--horizontal test ax-space--x0"
/>
`;

exports[`Separator renders with direction horizontal 1`] = `
<hr
  className="ax-separator ax-separator--solid ax-separator--horizontal ax-space--x0"
/>
`;

exports[`Separator renders with direction vertical 1`] = `
<hr
  className="ax-separator ax-separator--solid ax-separator--vertical ax-space--x0"
/>
`;

exports[`Separator renders with style dotted 1`] = `
<hr
  className="ax-separator ax-separator--dotted ax-separator--horizontal ax-space--x0"
/>
`;

exports[`Separator renders with style solid 1`] = `
<hr
  className="ax-separator ax-separator--solid ax-separator--horizontal ax-space--x0"
/>
`;
